src/main/scala/dotty/dokka/tasty/ClassLikeSupport.scala

```diff
@@ -12,87 +12,186 @@ trait ClassLikeSupport:
+        /*constuctors =*/ classDef.getConstructors.map(parseMethod(_)).asJava,
+        /*methods =*/ classDef.getMethods.map(parseMethod(_)).asJava,
+        /*fields =*/ (classDef.getTypeDefs.map(parseTypeDef) ++ classDef.getValDefs.map(parseValDef)).asJava,
+        /*nested =*/ (classDef.getNestedClasslikes).asJava,
         /*sources =*/ classDef.symbol.source,
         /*visibility =*/ sourceSet.asMap(classDef.symbol.getVisibility()),
         /*companion =*/ null,
-        /*generics =*/ typeParams.map(parseTypeArgument).asJava,
+        /*generics =*/ classDef.getTypeParams.map(parseTypeArgument).asJava,
         /*supertypes =*/ Map.empty.asJava, // Not used
         /*documentation =*/ classDef.symbol.documentation,
         /*expectPresentInSet =*/ null, // unused
         /*modifier =*/ sourceSet.asMap(modifier),
         inspector.sourceSet.toSet,
         PropertyContainer.Companion.empty()
-          .plus(ClasslikeExtension(parents, constructorMethod, kind, companion, parsedExtensions))
+          .plus(ClasslikeExtension(classDef.getParents, constructorMethod, Kind.Enum, classDef.getCompanion, classDef.getExtensionGroups))
+          .plus(AdditionalModifiers(sourceSet.asMap(extraModifiers.asJava)))
+          .plus(EnumExtension(enumVals ++ enumTypes ++ enumNested))
+      )
+
+  def parseTrait(classDef: reflect.ClassDef)(using ctx: Context): DClass = 
+    val constructorMethod = Some(parseMethod(classDef.constructor.symbol, constructorWithoutParamLists(classDef), s => classDef.getParameterModifier(s)))
+    new DClass(
+        classDef.symbol.dri,
+        classDef.name,
+        /*constuctors =*/ classDef.getConstructors.map(parseMethod(_)).asJava,
+        /*methods =*/ classDef.getMethods.map(parseMethod(_)).asJava,
+        /*fields =*/ (classDef.getTypeDefs.map(parseTypeDef) ++ classDef.getValDefs.map(parseValDef)).asJava,
+        /*nested =*/ classDef.getNestedClasslikes.asJava,
+        /*sources =*/ classDef.symbol.source,
+        /*visibility =*/ sourceSet.asMap(classDef.symbol.getVisibility()),
+        /*companion =*/ null,
+        /*generics =*/ classDef.getTypeParams.map(parseTypeArgument).asJava,
+        /*supertypes =*/ Map.empty.asJava, // Not used
+        /*documentation =*/ classDef.symbol.documentation,
+        /*expectPresentInSet =*/ null, // unused
+        /*modifier =*/ sourceSet.asMap(classDef.symbol.getModifier()),
+        inspector.sourceSet.toSet,
+        PropertyContainer.Companion.empty()
+          .plus(ClasslikeExtension(classDef.getParents, constructorMethod, Kind.Trait, classDef.getCompanion, classDef.getExtensionGroups))
+          .plus(AdditionalModifiers(sourceSet.asMap(classDef.symbol.getExtraModifiers().asJava)))
+      )
+
+
+  def parseClass(classDef: reflect.ClassDef)(using ctx: Context): DClass =
+    val constructorMethod = Some(parseMethod(classDef.constructor.symbol, constructorWithoutParamLists(classDef), s => classDef.getParameterModifier(s)))
+    new DClass(
```

Comment:
I think we need a helper function to create DClass instances

Reply:
did some refactor